src/lib/layer.js

```diff
@@ -409,6 +413,17 @@ export default class Layer {
 
   // Calculates uniforms
   drawLayer({uniforms = {}}) {
+    const {gl} = this.context;
+
+    // Apply polygon offset to avoid z-fighting
+    if (this.props.getPolygonOffset) {
+      gl.enable(GL.POLYGON_OFFSET_FILL);
```

Comment:
Whatever is being set here only applies to the following `layer.draw()` call. The offset mode can be overwritten in the layer's `draw` method if needed.